OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -296,6 +296,14 @@ public WDist GetMaximumRangeVersusTarget(Target target)
 		// Enumerates all armaments, that this actor possesses, that can be used against Target t
 		public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack)
 		{
+			// If the target is frozen, make sure it is still valid
+			if (t.Type == TargetType.FrozenActor)
+			{
+				var frozenLayer = self.Owner.PlayerActor.TraitOrDefault<FrozenActorLayer>();
+				if (frozenLayer == null || frozenLayer.FromID(t.FrozenActor.ID) == null)
```

Comment:
The `Target` is created in `ResolveFrozenActorOrder` and then used for this actor only through all the attack code.  The `FrozenActorLayer` used here and in `ResolveFrozenActorOrder` will be the same so long as the owner hasn't changed between the target creation and now.

We are *supposed* to be invalidating the actor's target when its owner changes, so don't need to deal with the more complicated case.  I can see where we do this for `AttackFollow` and subclasses, but not for the others.  If we really aren't cancelling targets when the owner changes for e.g. an infantry in any mod (`AttackFrontal`), or the TD obelisk (`AttackOmni`) then that's a bug that will need to be fixed (in its own pr, not here!).